boot/boot.go

```diff
@@ -143,29 +142,78 @@ func applicable(s snap.PlaceInfo, t snap.Type, dev Device) bool {
 	return true
 }
 
-// InUse checks if the given name/revision is used in the
-// boot environment
-func InUse(name string, rev snap.Revision) bool {
-	bootloader, err := bootloader.Find("", nil)
-	if err != nil {
-		logger.Noticef("cannot get boot settings: %s", err)
-		return false
+// bootState exposes the boot state for a type of boot snap.
+type bootState interface {
+	// revisions retrieves the revisions for the current snap
+	// revision (which always be set) and the try snap revision
+	// (which might be not set). it also returns whether the snap
+	// is in "trying" status.
```

Comment:
How about "revisions retrieves the revisions of the current snap and the try snap (only the latter might not be set), and whether the snap is in "trying" state"?
Otherwise, at least: drop the "revisions for the" at the start, or the "revision" after each thing; also add "will" before "be" in "always be set".